src/Cookie/SetCookie.php

```diff
@@ -384,9 +384,9 @@ public function validate()
                 . 'following characters: ()<>@,;:\"/?={}';
         }
 
-        // Value must not be empty, but can be 0
+        // Value must not be empty, but can be 0 or an empty string
         $value = $this->getValue();
-        if (empty($value) && !is_numeric($value)) {
+        if (empty($value) && !is_numeric($value) && $value !== '') {
```

Comment:
So "0" will still be invalid. 

I suggest modifying this if-statement to: `if ($value === null)`